### PR TITLE
Update JSF Dependencies to use Common Bundles 

### LINKED ISSUE #26946: Investigate and Update JSF / Faces Depenedencies if needed
Ensure our dependencies are using the latest releases. 
Latest Releases (As of Nov 21, 2023:
Commons BeanUtils: 1.9.4
Commons Collections: 3.2.2
Commons Discovery: .5 -- Is this even used? 
Commons Codec: 1.16.0
Commons Logging: 1.2
Commons Digester: 2.1

JSF 2.0:
See Closed Liberty Issue  `#30673`

[JSF 2.2:](https://github.com/OpenLiberty/open-liberty/blob/gm-23.0.0.11/dev/com.ibm.websphere.appserver.features/visibility/public/jsf-2.2/com.ibm.websphere.appserver.jsf-2.2.feature#L43-L52)
```
-bundles=com.ibm.ws.org.apache.commons.beanutils.1.9.4, \
 com.ibm.ws.org.apache.commons.collections, \
 com.ibm.ws.jsf.2.2, \
 com.ibm.ws.jsf.shared, \
 com.ibm.ws.jsf.jsp.shared, \
 com.ibm.ws.org.apache.commons.discovery.0.2, \
 io.openliberty.org.apache.commons.codec, \
 io.openliberty.org.apache.commons.logging, \
 com.ibm.ws.cdi.interfaces, \
 com.ibm.websphere.appserver.thirdparty.jsf-2.2; location:="dev/api/third-party/"
```
https://github.com/OpenLiberty/open-liberty/blob/ea7a8d8ca00a2b02380da3fc850771e918af7bd2/dev/com.ibm.ws.jsf.2.2/bnd.bnd#L163

[JSF 2.3:](https://github.com/OpenLiberty/open-liberty/blob/gm-23.0.0.11/dev/com.ibm.websphere.appserver.features/visibility/public/jsf-2.3/com.ibm.websphere.appserver.jsf-2.3.feature#L48-L57)
```
-bundles=com.ibm.ws.org.apache.myfaces.2.3, \
 com.ibm.ws.org.apache.commons.beanutils.1.9.4, \
 com.ibm.ws.org.apache.commons.collections, \
 com.ibm.ws.org.apache.commons.discovery.0.2, \
 io.openliberty.org.apache.commons.logging, \
 com.ibm.ws.jsf.shared, \
 com.ibm.ws.jsf.jsp.shared, \
 io.openliberty.faces.internal, \
 com.ibm.ws.cdi.interfaces, \
 com.ibm.websphere.appserver.thirdparty.jsf-2.3; location:="dev/api/third-party/"
 ```
 https://github.com/OpenLiberty/open-liberty/blob/ea7a8d8ca00a2b02380da3fc850771e918af7bd2/dev/com.ibm.ws.org.apache.myfaces.2.3/bnd.bnd#L137
 
[ faces-3.0:](https://github.com/OpenLiberty/open-liberty/blob/gm-23.0.0.11/dev/com.ibm.websphere.appserver.features/visibility/public/faces-3.0/io.openliberty.faces-3.0.feature#L46-L56)
```
-bundles=io.openliberty.org.apache.myfaces.3.0, \
 com.ibm.ws.org.apache.commons.beanutils.1.9.4, \
 com.ibm.ws.org.apache.commons.collections, \
 com.ibm.ws.org.apache.commons.discovery.0.2, \
 io.openliberty.org.apache.commons.logging, \
 com.ibm.ws.jsf.shared.jakarta, \
 com.ibm.ws.jsf.jsp.shared.jakarta, \
 io.openliberty.faces.internal.jakarta, \
 com.ibm.ws.cdi.interfaces.jakarta, \
 io.openliberty.jakarta.jstl.2.0; location:="dev/api/spec/,lib/"; mavenCoordinates="jakarta.servlet.jsp.jstl:jakarta.servlet.jsp.jstl-api:2.0.0", \
 io.openliberty.faces.3.0.thirdparty; location:="dev/api/third-party/"
 ```
 https://github.com/OpenLiberty/open-liberty/blob/ea7a8d8ca00a2b02380da3fc850771e918af7bd2/dev/io.openliberty.org.apache.myfaces.3.0/bnd.bnd#L125

### PR DESCRIPTION
**Pull request by volosied**
for #26946 

Highlights of Changes:
- Created io.openliberty.org.apache.commons.digester
  - Digester project was created so that JSF 2.0 can use this bundle rather than packaging the code within the JSF bundle.
  - Removed the commons codec bundle from JSF 2.3 and 3.0 since it's not used (code was refactored to use java.util.base64 instead)
  - Note: Commons Collections is still kept since it's used by bean utils.

Verified:
-  Commons Projects use latest features (Note: Codec's latest release is now on 1.16.0, but 1.15 is currently used in OL.  See [Release Notes of Changes](https://commons.apache.org/proper/commons-codec/changes-report.html#a1.16.0))

If any dependency updates are required from now on, we can just update the bundle projects. 

TODO:
- Delete com.ibm.ws.org.apache.commons.discovery since I didn't see it used and all our tests pass without it.  (No combined builds can be done due to holidays, so this will be in a separate PR). See #27160

### CODE REVIEW

**Comment by volosied** [comment]
Combined Personal Build with OL / CD Branches: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_zMRrwI32Ee6gttl9kBC4bw&tab=com.ibm.team.build.web.ui.internal.editors.result.test.TestProvider

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsf.2.3_fat,com.ibm.ws.jsf.2.2_fat

File: dev/io.openliberty.org.apache.commons.digester/bnd.overrides [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-include= ~../cnf/resources/bnd/rejar.props
+bVersion=1.15
```
Comment (volosied) [inline dev/io.openliberty.org.apache.commons.digester/bnd.overrides]:
```suggestion
#*******************************************************************************
# Copyright (c) 2023 IBM Corporation and others.
# All rights reserved. This program and the accompanying materials
# are made available under the terms of the Eclipse Public License 2.0
# which accompanies this distribution, and is available at
# http://www.eclipse.org/legal/epl-2.0/
# 
# SPDX-License-Identifier: EPL-2.0
#*******************************************************************************
-include= ~../cnf/resources/bnd/rejar.props
bVersion=1.1
```

File: dev/io.openliberty.org.apache.commons.digester/.project [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>com.ibm.ws.org.apache.commons.discovery</name>
+	<name>io.openliberty.org.apache.commons.codec</name>
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.commons.digester/.project]: Shouldn't this be `io.openliberty.org.apache.commons.digester`?

File: dev/io.openliberty.org.apache.commons.digester/bnd.bnd [hunk 3]
```diff
@@ -10,9 +10,6 @@
 # Contributors:
 #     IBM Corporation - initial API and implementation
 #*******************************************************************************
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.commons.digester/bnd.bnd:9]: Clean up Contributors in the copyright

File: dev/io.openliberty.org.apache.commons.digester/bnd.overrides [hunk 4]
```diff
@@ -8,7 +8,7 @@
 # SPDX-License-Identifier: EPL-2.0
 #*******************************************************************************
 -include= ~../cnf/resources/bnd/rejar.props
-bVersion=1.15
+bVersion=2.1
```
Comment (volosied) [inline dev/io.openliberty.org.apache.commons.digester/bnd.overrides:11]: Other bundles match the version of the jar, so I did the same here.

**Review by pnicolucci — DISMISSED** [review]

**Comment by volosied** [comment]
#build

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby

**Pull request merged**